config/crd/components/all/more_validations_in_bindingpolicies.yaml

```diff
@@ -0,0 +1,12 @@
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/clusterSelectors/items/properties/matchLabels/additionalProperties/pattern
+  value: "^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$"
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/clusterSelectors/items/properties/matchExpressions/items/properties/key/pattern
+  value: "^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$"
```

Comment:
This should be the restriction for label names, not label values.

Reply:
Yes. Label names are validated here. And they are validated using the same regex as the one used by the apiserver.
 
```
The Deployment "nginx-deployment" is invalid: 
* spec.selector.matchExpressions[0].key: Invalid value: "-tier": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
* spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"nginx"}, MatchExpressions:[]v1.LabelSelectorRequirement{v1.LabelSelectorRequirement{Key:"-tier", Operator:"In", Values:[]string{"cache", "local"}}}}: invalid label selector
```

Reply:
Turns out that the key's _prefix_ should also be validated. Added this to the regex. Added/modified unit test cases for the key's prefix.